pennylane/resource/resource.py

```diff
@@ -71,6 +75,35 @@ def _ipython_display_(self):
         print(str(self))
 
 
+class OperationWithResources(Operation):
```

Comment:
To discuss the naming - I feel like `ResourceOp` is shorter and still descriptive.

Reply:
I agree we should shorten it. To me `ResourceOp` sounds more like a new custom operation than a base class to inherit from.

Reply:
`ResourceOperation`?

Reply:
`ResourceOperation` sounds good to me.